### PR TITLE
Feature/devsu 1469 ref alt in tgr

### PR DESCRIPTION
**Pull request by Anna-log7**
This PR adds in ref/alt data from the small mutation endpoint to the genomic events table in the probe summary.
I refactored the PrintTable component to generalize its use as well. This is due to ag-grid not handling long amounts of text/sizing issues in the print view with ag-grid tables.

### CODE REVIEW

**Review by kttkjl — COMMENTED** [review]
~~Weird, did the API change? `/me` shows me a 400 telling me `message: "Invalid authorization token"~~

Looks fixed now

File: app/components/PrintTable/index.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+              {columnDefs.sort(headerSortFunc).map((col) => (
+                <th key={col.headerName}>
+                  {col.headerName}
+                </th>
+              ))}
+            </tr>
+          </thead>
+          <tbody>
+            {data.map((val, index) => (
+              <tr key={index} className="table__row">
```
[the remark this answers is not in the document]
Comment (Anna-log7) [reply app/components/PrintTable/index.tsx:80]: There's no common key, no

File: app/views/ReportView/components/ProbeSummary/columnDefs.ts [hunk 2]
```diff
@@ -1,31 +1,37 @@
 const sampleColumnDefs = [
   {
     headerName: 'Sample',
+    colId: 'Sample',
```
[the remark this answers is not in the document]
Comment (Anna-log7) [reply app/views/ReportView/components/ProbeSummary/columnDefs.ts:4]: Yes, but I access the colId from columnDefs in the printTable for sorting so it needs to be defined initially

File: app/views/ReportView/components/ProbeSummary/index.tsx [hunk 3]
```diff
@@ -10,25 +10,23 @@ import EditIcon from '@material-ui/icons/Edit';
...
 import PatientEdit from '../GenomicSummary/components/PatientEdit';
 import EventsEditDialog from './components/EventsEditDialog';
-import PrintTable from './components/PrintTable';
+import ProbeResultsType from './types.d';
 
 import './index.scss';
 
 type ProbeSummaryProps = {
-  loadedDispatch: (type: Record<string, unknown>) => void;
+  loadedDispatch: ({ 'type': string }) => void;
```
[the remark this answers is not in the document]
Comment (Anna-log7) [reply app/views/ReportView/components/ProbeSummary/index.tsx:29]: Nope, that's right haha
It should pass `{ type: 'typehere' }`

File: app/components/PrintTable/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (kttkjl) [reply app/components/PrintTable/index.tsx:80]: Seeing a load of errors on react calling out the keys, but I suppose if they're not needing updates I suppose it's .... fine? 

Perhaps you can have a prop here that specifies the `name` of this specific `PrintTable`, then with index and then the actual row name

[the remark this answers is not in the document]
Comment (kttkjl) [reply app/components/PrintTable/index.tsx:80]: Actually yeah, if you're generalizing a PrintTable perhaps it's best to give a prop telling WHICH PrintTable it is

**Review by kttkjl — CHANGES_REQUESTED** [review]
Add prop to PrintTable to allow for uniquely identifying each PrintTable

File: app/components/PrintTable/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Anna-log7) [reply app/components/PrintTable/index.tsx:80]: The key errors are in a different component in the appendices unfortunately. The line you commented about is per row of the table, so I don't think a `name` for the table itself would solve the issue here

[the remark this answers is not in the document]
Comment (kttkjl) [reply app/components/PrintTable/index.tsx:80]: flagged wrong line, key should be ok here

ran into problems with `in table (created by ConfigTable)` having the <tr> having "" as key

**Review by kttkjl — APPROVED** [review]

**Pull request merged**